semgrep/semgrep/metric_manager.py

```diff
@@ -46,8 +46,23 @@ def __init__(self) -> None:
 
         self._send_metrics = False
 
-    def set_project_hash(self, project_hash: Optional[str]) -> None:
-        self._project_hash = project_hash
+    def set_project_hash(self, project_url: Optional[str]) -> None:
+        """
+        Standardizes url then hashes
+        """
+        if project_url is None:
+            self._project_hash = None
+        else:
+            parsed_url = urlparse(project_url)
+            if parsed_url.scheme == "https":
+                # Remove optional username/password from project_url
+                sanitized_url = f"{parsed_url.hostname}{parsed_url.path}"
```

Comment:
Should we include other parsed URL information? E.g. `params`, `query`, `fragment`, or `port`?

Since we're generalizing the solution to all possible `git` repositories, and not just Gitlab/Github, there could be all kinds of URLs returned, not just what we've come to expect from those SCMs.

I suspect the failure mode of _not_ including this other information is that some weird SCM somewhere will have duplicate project hashes because the hostname + paths are all the same, and it uses the query string for lookup, or something like that. Basically, we over count a particular project hash because we avoided additional context.

The failure mode of including additional information seems to be the issue we faced originally: we accidentally include information that makes the metric useless. Then we have under counting instead of the over counting mentioned above.

It's hard to say what all the weird types of URLs we might receive are. Or the weird `git` authentication schemes leveraging this URL. We want to get this right insofar as we can, though, because every time we change this we either invalidate the old data, or make it difficult to connect the old and new.

What do you think?

Reply:
I think you're right to be broad, but not so broad as to cause us to create differences between different runs as the same repo. @brendongo do we have a sense of where that line lines for gitlab? wondering if you're excluding query for example b/c that might change run-to-run?

Reply:
I think we prefer to undercount instead of possiibly overcounting. For now none of the urls we care about seem to use these extra url information so the risk of invalidating old data/not being able to connect old and new is low